lib/Dialect/FIRRTL/Import/FIRAnnotations.cpp

```diff
@@ -19,6 +20,43 @@
 
 namespace json = llvm::json;
 
+using namespace circt;
+using namespace firrtl;
+
+/// Given a string \p target, starting with either '.' or '[', this function
+/// splits the string at every '[' and '.' and populates the \p annotations with
+/// the array of strings. Assumption, the \p target string is a well formed
+/// valid token specifying an instance of a bundle/array.
+void static parseSubFieldSubIndexAnnotations(StringRef target,
+                                             ArrayAttr &annotations,
+                                             MLIRContext *context) {
+  std::string temp = "";
+  if (target.empty())
+    return;
+  char begin = target[0];
+  temp.push_back(begin);
+  SmallVector<Attribute> annotationVec;
+  // The caller must strip the prefix, and the string target must only contain
+  // the suffix.
+  if (begin != '.' && begin != '[')
+    return;
+  for (size_t i = 1; i < target.size(); i++) {
+    if (target[i] == '[') {
+      // Create a StringAttr with the previous token.
+      annotationVec.push_back(StringAttr::get(context, temp));
+      temp = "";
```

Comment:
I'd recommend `temp.clear()` instead of assigning "".  It is simpler and more efficient.